DiztinGUIsh/static/Project.cs

```diff
@@ -215,107 +247,54 @@ private static void OpenVersion0(byte[] unzipped, OpenFileDialog open)
-                for (int i = 0; i < size; i++) Data.SetDirectPage(i, unzipped[pointer + size + i] | (unzipped[pointer + 2 * size + i] << 8));
-                for (int i = 0; i < size; i++) Data.SetXFlag(i, unzipped[pointer + 3 * size + i] != 0);
-                for (int i = 0; i < size; i++) Data.SetMFlag(i, unzipped[pointer + 4 * size + i] != 0);
-                for (int i = 0; i < size; i++) Data.SetFlag(i, (Data.FlagType)unzipped[pointer + 5 * size + i]);
-                for (int i = 0; i < size; i++) Data.SetArchitechture(i, (Data.Architechture)unzipped[pointer + 6 * size + i]);
-                for (int i = 0; i < size; i++) Data.SetInOutPoint(i, (Data.InOutPoint)unzipped[pointer + 7 * size + i]);
-
-                pointer += 8 * size;
-                int label_count = Util.ByteArrayToInteger(unzipped, pointer);
+                int offset = converter(Util.ByteArrayToInteger(unzipped, pointer));
                 pointer += 4;
 
-                AliasList.me.Reset();
-                for (int i = 0; i < label_count; i++)
-                {
-                    int offset = Util.ByteArrayToInteger(unzipped, pointer);
-                    pointer += 4;
+                var comment = Util.ReadZipString(unzipped, ref pointer);
 
-                    string label = "";
-                    while (unzipped[pointer] != 0) label += (char)unzipped[pointer++];
-                    pointer++;
+                Data.AddComment(offset, comment, true);
+            }
+        }
 
-                    Data.AddLabel(offset, label, true);
-                }
+        private static void ReadAliases(byte[] unzipped, ref int pointer, AddressConverter converter, bool readAliasComments)
+        {
+            int count = Util.ByteArrayToInteger(unzipped, pointer);
+            pointer += 4;
 
-                int comment_count = Util.ByteArrayToInteger(unzipped, pointer);
+            for (int i = 0; i < count; i++)
+            {
+                int offset = converter(Util.ByteArrayToInteger(unzipped, pointer));
                 pointer += 4;
 
-                for (int i = 0; i < comment_count; i++)
-                {
-                    int offset = Util.ByteArrayToInteger(unzipped, pointer);
-                    pointer += 4;
+                var aliasInfo = new Data.AliasInfo {
+                    name = Util.ReadZipString(unzipped, ref pointer), 
+                    comment = readAliasComments ? Util.ReadZipString(unzipped, ref pointer) : "",
+                };
+                aliasInfo.CleanUp();
 
-                    string comment = "";
-                    while (unzipped[pointer] != 0) comment += (char)unzipped[pointer++];
-                    pointer++;
-
-                    Data.AddComment(offset, comment, true);
-                }
-            }
-            else
-            {
-                throw new Exception("Couldn't open the ROM file!");
+                Data.AddLabel(offset, aliasInfo, true);
```

Comment:
someone else's PR mentioned using some kind of BinaryReader/Writer, I agree with that.

unrelated to this PR, I wonder if doing some kind of text-ish format would produce prohibitively gigantic files.  thinking about a use case for multiple collaborators and having a hope of merging data together.